src/execution_plan/ops/op_expand_into.c

```diff
@@ -264,23 +210,15 @@ static void ExpandIntoFree(OpBase *ctx) {
 		op->M = GrB_NULL;
 	}
 
-	if(op->edges) {
-		array_free(op->edges);
-		op->edges = NULL;
-	}
-
-	if(op->edgeRelationTypes) {
-		array_free(op->edgeRelationTypes);
-		op->edgeRelationTypes = NULL;
-	}
-
 	if(op->ae) {
 		AlgebraicExpression_Free(op->ae);
 		op->ae = NULL;
 	}
 
+	if(op->setEdge) Traverse_FreeEdgeData(&op->edge_data);
```

Comment:
I think that it is better to have a pointer to an allocated struct.
Call the free function regardless and remove line 189 as it is breaking abstraction